.cbmc-batch/include/proof_helpers/aws_byte_cursor_read_common.h

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/common/byte_buf.h>
+#include <proof_helpers/make_common_data_structures.h>
+#include <proof_helpers/proof_allocators.h>
+
+void aws_byte_cursor_read_common_harness() {
+    /* parameters */
+    struct aws_byte_cursor cur;
+    size_t length;
+    DEST_TYPE *dest = can_fail_malloc(length);
+
+    /* assumptions */
+    ensure_byte_cursor_has_allocated_buffer_member(&cur);
+    __CPROVER_assume(aws_byte_cursor_is_valid(&cur));
+    __CPROVER_assume(AWS_MEM_IS_READABLE(cur.ptr, BYTE_WIDTH));
+    __CPROVER_assume(AWS_OBJECT_PTR_IS_WRITABLE(dest));
+
+    /* save current state of the data structure */
+    struct aws_byte_cursor old_cur = cur;
+    struct store_byte_from_buffer old_byte_from_cur;
+    save_byte_from_array(cur.ptr, cur.len, &old_byte_from_cur);
+    DEST_TYPE dest_copy;
+    memcpy(&dest_copy, old_cur.ptr, BYTE_WIDTH);
+    dest_copy = AWS_NTOH(dest_copy);
+
+    /* operation under verification */
+    if (BYTE_CURSOR_READ(&cur, dest)) {
```

Comment:
I assume this returns success/error? ITs cleaner to check for the actual value, rather than have to remember which way the if goes.

Reply:
It does not return an error code, it returns a Boolean variable. Each proof harness defines `BYTE_CURSOR_READ` w.r.t. the function under verification. For instance, the proof harness for  `aws_byte_cursor_read_be16` defines the following:
```
#define BYTE_CURSOR_READ aws_byte_cursor_read_be16
```